### PR TITLE
Added introspect for scroll view in TabBarView with PageTabViewStyle

### PR DESCRIPTION
**Pull request by philprime**
I created an extension to introspect the horizontal scroll view in a `TabBarView` with `tabViewStyle(PageTabViewStyle()`.

This tab view style is Apple's solution for pagination like UIPageViewController. Interesting enough, it actually uses a `UICollectionView` with an embedded `UIScrollView`.

I use this extension to disable horizontal scrolling (similar to `scrollView.isScrollEnabled = false`) so only programmatic navigation is possible. When horizontal scrolling is disabled, the collection view still bounces vertically, so both views are passed to the customize closure.

**Example:**

```swift
import SwiftUI
import Introspect

@main
struct DemoApp: App {

    @State private var selection = 1

    var body: some Scene {
        WindowGroup {
            TabView(selection: $selection) {
                Button("Go To Page 2", action: {
                    withAnimation {
                        selection = 2
                    }
                })
                .tag(1)
                Button("Go To Page 1", action: {
                    withAnimation {
                        selection = 1
                    }
                })
                .tag(2)
            }
            .tabViewStyle(PageTabViewStyle())
            .introspectPagedTabView { collectionView, scrollView in
                collectionView.isScrollEnabled = false
                scrollView.isScrollEnabled = false
            }
        }
    }
}
```

### CODE REVIEW

**Comment by dannyhertz** [comment]
+1 can we get this merged?

**Comment by joshuakcockrell** [comment]
This is working great for me! Setting the `collectionView ` or `scrollView` delegate methods breaks the `TabBar` though.. SwiftUI must be doing some magic behind the scenes with the delegates.

**Review by guillecom — APPROVED** [review]
Tested this and it works in ios 15.4

**Comment by SplittyDev** [comment]
@philprime thanks a lot for the PR!
I'd love to merge this, could you add tests to make sure this works as intended on all platforms?

**Comment by philprime** [comment]
@SplittyDev I think I can do that, but I need assistance on how to create Introspect tests cases -> how do you verify it works?

**Comment by SplittyDev** [comment]
> @SplittyDev I think I can do that, but I need assistance on how to create Introspect tests cases -> how do you verify it works?

In the IntrospectTests directory we have `AppKitTests.swift` and `UIKitTests.swift`. If an introspection works for both, we write two separate tests for AppKit and UIKit, otherwise we just write one for the correct platform.

We create a view for the test (e.g. `TabBarTestView` / `PageTabViewStyleTestView`), where the target component is introspected and a function (called a `spy`) is called from within the introspection.

We then write the test itself, which renders the test view and passes an `XCTestExpectation` to the component as the `spy` function. Then we present the component using a small helper utility and verify that the `spy` was called.

One more important thing is that the components / introspections are properly tagged with `@available` annotations, here is an example from the `ColorWellTestView`:
```swift
@available(iOS 14.0, tvOS 13.0, macOS 11.0, *)
@available(tvOS, unavailable)
private struct ColorWellTestView: View { ... }
```

Platform availability information can usually be taken directly from the docs:
Apple Docs: https://developer.apple.com/documentation/swiftui/pagetabviewstyle

File: IntrospectTests/UIKitTests.swift [hunk 1]
```diff
@@ -364,17 +382,17 @@ class UIKitTests: XCTestCase {
         TestUtils.present(view: view)
         wait(for: [expectation], timeout: TestUtils.Constants.timeout)
     }
-    
+
```
Comment (philprime) [inline IntrospectTests/UIKitTests.swift:385]:
```suggestion
```

File: IntrospectTests/UIKitTests.swift [hunk 2]
```diff
@@ -364,17 +382,17 @@ class UIKitTests: XCTestCase {
...
     func testTabViewRoot() {
-        
+
```
Comment (philprime) [inline IntrospectTests/UIKitTests.swift:387]:
```suggestion
```

File: IntrospectTests/UIKitTests.swift [hunk 3]
```diff
@@ -364,17 +382,17 @@ class UIKitTests: XCTestCase {
...
         let expectation = XCTestExpectation()
         let view = TabRootTestView(spy: {
             expectation.fulfill()
         })
         TestUtils.present(view: view)
         wait(for: [expectation], timeout: TestUtils.Constants.timeout)
     }
-    
+
```
Comment (philprime) [inline IntrospectTests/UIKitTests.swift:395]:
```suggestion
```

**Comment by philprime** [comment]
@SplittyDev I added a test case as you described it. Let me know if more changes are required.

(Sorry for the formatting commits/notifications, for some reason it kept being wrong 😂)

**Comment by SplittyDev** [comment]
@philprime good stuff, thanks a lot! The formatting commits aren't a problem, we'll squash-merge anyway :)

**Comment by ichikmarev** [comment]
@SplittyDev Hi. This awesome PR was merged but seemed lost since it is not in 0.1.4.

**Comment by philprime** [comment]
@ichikmarev actually the PR hasn't been released yet.

My commit is in the main branch with date April 17th, 2022, but the release 0.1.4 was published on January 26, 2022

**Pull request merged**